src/oscar/models/fields/__init__.py

```diff
@@ -73,14 +74,31 @@ class UppercaseCharField(CharField):
     Defined with the with_metaclass helper so that to_python is called
     https://docs.djangoproject.com/en/1.6/howto/custom-model-fields/#the-subfieldbase-metaclass  # NOQA
     """
-
     def contribute_to_class(self, cls, name, **kwargs):
         super().contribute_to_class(
             cls, name, **kwargs)
         setattr(cls, self.name, Creator(self))
 
-    def from_db_value(self, value, expression, connection, context):
-        return self.to_python(value)
+    # Django 2.0 updates the signature of from_db_value.
+    # https://docs.djangoproject.com/en/2.0/releases/2.0/#context-argument-of-field-from-db-value-and-expression-convert-value
+    if django.VERSION < (2,):
+        def from_db_value(self, value, expression, connection, context):
```

Comment:
I don't think we need to have different definitions based on Django version. We can simplify our code to:

```
def from_db_value(self, value, *args, **kwargs):
```
Since we don't use any of the other args. This will work for all versions of Django.